ILRepack/Steps/SourceServerData/SourceServerDataRepackStep.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using SourceLink;
+
+namespace ILRepacking.Steps.SourceServerData
+{
+    internal class SourceServerDataRepackStep : IRepackStep
+    {
+        private readonly string _targetPdbFile;
+
+        private readonly IEnumerable<string> _assemblyFiles;
+
+        private byte[] _srcSrv;
+
+        public SourceServerDataRepackStep(string targetFile, IEnumerable<string> assemblyFiles)
+        {
+            if (targetFile == null) throw new ArgumentNullException(nameof(targetFile));
+            if (assemblyFiles == null) throw new ArgumentNullException(nameof(assemblyFiles));
+            _targetPdbFile = PdbPath(targetFile);
+            _assemblyFiles = assemblyFiles;
+        }
+
+        public void Perform()
+        {
+            if (_assemblyFiles.Any())
+            {
+                var pdbs = _assemblyFiles.Select(PdbPath).ToArray();
+                string primaryPdb = pdbs.First();
+                if (File.Exists(primaryPdb))
+                {
+                    _srcSrv = PdbFile.readSrcSrvBytes(primaryPdb);
+                    HttpSourceServerDescriptor primaryDescriptor;
+                    if(HttpSourceServerDescriptor.TryParse(_srcSrv, out primaryDescriptor))
```

Comment:
I make manadatory to have a primary srcsrv because, in the case no primary has it and we are merging srcsrv with different "versionControl", we will take arbitrary the one of the first which has it. I didn't feel comfortable with that.
In a usage point of view, if the user doesn't have a PDB or srcsrv in his primary library, why should he care to have it generated ? But it's debattable. Merging whatever we have a primary or not would make probably the code a bit more simple.